openedx/core/djangoapps/schedules/README.rst

```diff
@@ -0,0 +1,366 @@
+Dynamic Pacing Schedules
+========================
+
+The Schedules app allows course teams to automatically email learners in
+self-paced courses. The emails are designed to keep learners engaged
+with a course's content. Learners receive these messages at important
+milestones throughout a course.
+
+With Schedules, the author of a self-paced course opts learners into one
+of two "Schedule Experiences". Learners either receive Weekly Course
+Highlight Messages, or a combination of Recurring Nudges and Upgrade
+Reminders.
+
+The app can send all three message types "out of the box":
+
+-  Recurring Nudges
+-  Upgrade Reminders
+-  Weekly Course Highlight Messages
+
+Recurring Nudges encourage learners to engage with self-paced courses at
+regular intervals. The app sends nudges three days and ten days after a
+learner enrolls in a course.
+
+Upgrade Reminders ask learners to purchase their course's "Verified"
+certificate. The reminders are sent two days from their course's upgrade
+deadline, or two days from the course's end date. Whichever date occurs
+sooner.
+
+Weekly Course Highlight Messages tell learners what to look forward to
+in the coming week of a course. Course authors provide "section
+highlights" when authoring a course in Studio. The app generates emails
+with these section highlights listed in the body of the message.
+
+The app introduces the Schedule object to the edX codebase. Learners
+receive Schedules when they enroll in self-paced courses. With
+Schedules, the app determines when to send particular messages. It
+assumes a steady rate of progress through course materials. It
+determines the cadence of its messaging through a course's "Schedule
+Experience".
+
+Glossary
+--------
+
+-  **Schedule**: Stores the day a learner enrolls in a course and the
+   learner's "upgrade deadline".
+
+-  **Schedule Experience**: Defines which set of emails that a learner
+   will receive so that they have a consistent email experience
+   throughout their time in the course even if a course author makes
+   changes to the course after they enroll. Right now, the possible
+   Schedule Experiences are:
+
+   -  "Recurring Nudge and Upgrade Reminder"
+   -  "Course Updates"
+
+-  **Upgrade Deadline**: The date before which a learner can purchase a
+   verified certificate. A Schedule imposes a "soft" upgrade deadline 21
+   days from when a learner enrolled in a course. A self-paced course
+   imposes a "hard" upgrade deadline that is the course-wide expiration
+   date for upgrading on the course. A learner's Schedule will use
+   whichever date is earlier.
+
+-  **Course Update**: In the code, we refer to Weekly Course Highlight
+   messages as Course Updates. In hindsight, this is confusing since we
+   also use the term Course Updates to refer to bulk-emails manually
+   sent out by course instructors. We haven't had the chance to rename
```

Comment:
Course Updates *also* refers to the blocks of text that appear at the top of the course outline...